src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/BlackDuckBomEditCollector.java

```diff
@@ -67,91 +59,65 @@
 
     @Autowired
-    public BlackDuckBomEditCollector(final JsonExtractor jsonExtractor, final BlackDuckProperties blackDuckProperties) {
+    public BlackDuckBomEditCollector(JsonExtractor jsonExtractor, BlackDuckProperties blackDuckProperties) {
         super(jsonExtractor, List.of(BlackDuckContent.BOM_EDIT), blackDuckProperties);
     }
 
     @Override
-    protected List<LinkableItem> getTopicItems(final JsonFieldAccessor accessor, final List<JsonField<?>> fields) {
-        final JsonField<String> topicField = getDataField(fields, FieldContentIdentifier.TOPIC);
+    protected List<LinkableItem> getTopicItems(JsonFieldAccessor accessor, List<JsonField<?>> fields) {
+        JsonField<String> topicField = getDataField(fields, FieldContentIdentifier.TOPIC);
         return getItemFromProjectVersionWrapper(accessor, topicField, ProjectVersionWrapper::getProjectView,
             view -> new LinkableItem(topicField.getLabel(), ((ProjectView) view).getName(), view.getHref().orElse(null)));
     }
 
     @Override
-    protected List<LinkableItem> getSubTopicItems(final JsonFieldAccessor accessor, final List<JsonField<?>> fields) {
-        final JsonField<String> subTopicField = getDataField(fields, FieldContentIdentifier.SUB_TOPIC);
+    protected List<LinkableItem> getSubTopicItems(JsonFieldAccessor accessor, List<JsonField<?>> fields) {
+        JsonField<String> subTopicField = getDataField(fields, FieldContentIdentifier.SUB_TOPIC);
         return getItemFromProjectVersionWrapper(accessor, subTopicField, ProjectVersionWrapper::getProjectVersionView,
             view -> new LinkableItem(subTopicField.getLabel(), ((ProjectVersionView) view).getVersionName(), view.getHref().orElse(null)));
     }
 
     @Override
-    protected Collection<ComponentItem> getComponentItems(final JsonFieldAccessor jsonFieldAccessor, final List<JsonField<?>> notificationFields, final AlertNotificationWrapper notificationContent) {
-        final JsonField<String> topicField = getDataField(notificationFields, FieldContentIdentifier.CATEGORY_ITEM);
-        final Optional<String> bomComponentUrl = getBomComponentUrl(jsonFieldAccessor, topicField);
-        final Optional<VersionBomComponentView> versionBomComponentView = bomComponentUrl.flatMap(this::getBomComponentView);
-        final Optional<ProjectVersionWrapper> projectVersionWrapper = versionBomComponentView.flatMap(this::getProjectVersionWrapper);
+    protected Collection<ComponentItem> getComponentItems(JsonFieldAccessor jsonFieldAccessor, List<JsonField<?>> notificationFields, AlertNotificationWrapper notificationContent) {
+        JsonField<String> topicField = getDataField(notificationFields, FieldContentIdentifier.CATEGORY_ITEM);
+        Optional<String> bomComponentUrl = getBomComponentUrl(jsonFieldAccessor, topicField);
+        Optional<VersionBomComponentView> versionBomComponentView = bomComponentUrl.flatMap(url -> getBlackDuckDataHelper().getBomComponentView(url));
```

Comment:
If you store the BlackDuckDataHelper as a variable you can still used method references here.